xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -363,13 +363,24 @@ posix_lookup(call_frame_t *frame, xlator_t *this, loc_t *loc, dict_t *xdata)
 out:
     if (!op_ret && !gfidless && gf_uuid_is_null(buf.ia_gfid)) {
         gf_msg(this->name, GF_LOG_ERROR, ENODATA, P_MSG_NULL_GFID,
-               "buf->ia_gfid is null for "
-               "%s",
-               (real_path) ? real_path : "");
+               "buf->ia_gfid is null for %s",
+               (real_path) ? real_path : "(null)");
         op_ret = -1;
         op_errno = ENODATA;
     }
 
+    /* TODO: get the path */
+    /* In the full run of regression, I was not able to hit this case, hence
+       leaving it as TODO. Good to have logic of resolving GFID only access
+       to a path for many other features too. But initial version can just
+       be knowning that we are hitting the scenario in certain usecases */
+    if ((op_ret == 0) && (dict_get_sizen(xdata, "get-full-path"))) {
```

Comment:
Couldn't we reuse one of the already existing implementations ? I've found many ways to get something similar, like:

* trusted.glusterfs.pathinfo
* trusted.pathinfo
* glusterfs.gfid2path
* glusterfs.gfidtopath
* glusterfs.ancestry.path

Reply:
Yes. That was the plan. I have implemented this to check how many this requests I receive in posix. In last ~8 months in pushing it out with kadalu releases, I have seen 2 instances of this log, meaning, we need to fix it. But the frequency is not high. We need to take a call about should we allow gfid based lookup at all and then proceed IMO. For this PR, shall I keep it as is, as this is not an implementation which is complete. Or will keep just a TODO for now.

Reply:
That's fine for now.